sched/mutex/mutex_breaklock.c

```diff
@@ -0,0 +1,173 @@
+/****************************************************************************
+ * sched/mutex/mutex_breaklock.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/mutex.h>
+
+#include <errno.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nxmutex_breaklock
+ *
+ * Description:
+ *   This function attempts to break the mutex
+ *
+ * Parameters:
+ *   mutex   - Mutex descriptor.
+ *   locked  - Is the mutex break success
+ *
+ * Return Value:
+ *   This is an internal OS interface and should not be used by applications.
+ *   It follows the NuttX internal error return policy:  Zero (OK) is
+ *   returned on success.  A negated errno value is returned on failure.
+ *   Possible returned errors:
+ *
+ ****************************************************************************/
+
+int nxmutex_breaklock(FAR mutex_t *mutex, FAR bool *locked)
+{
+  int ret = OK;
+
+  *locked = false;
+  if (nxmutex_is_hold(mutex))
+    {
+      ret = nxmutex_unlock(mutex);
+      if (ret >= 0)
+        {
+          *locked = true;
+        }
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: nxrmutex_breaklock
+ *
+ * Description:
+ *   This function attempts to break the recursive mutex
+ *
+ * Parameters:
+ *   rmutex - Recursive mutex descriptor.
```

Comment:
```suggestion
 * Input Parameters:
 *   rmutex - Recursive mutex descriptor.
 *
 * Output Parameters:
 *   count  -
```
Missing documentation for `count`.